lib/router/router.js

```diff
@@ -117,22 +108,14 @@ export default class Router {
 
     this.abortComponentLoad()
 
-    let data
-    let props
-    let _err
-    try {
-      data = await this.fetchComponent(route)
-      const ctx = { ...data.ctx, pathname, query }
-      props = await this.getInitialProps(data.Component, ctx)
-    } catch (err) {
-      if (err.cancelled) return false
+    this.emit('routeChangeStart', route, url)
+    const {
+      data, props, error
+    } = await this.getRouteInfo(route, pathname, query)
 
-      data = { Component: this.ErrorComponent, ctx: { err } }
-      const ctx = { ...data.ctx, pathname, query }
-      props = await this.getInitialProps(data.Component, ctx)
-
-      _err = err
-      console.error(err)
+    if (error) {
+      this.emit('routeChangeError', error, route, url)
+      throw error
```

Comment:
Yep. Even abort is some kind of an error.
Here it won't show on the screen, but the user could catch them via events or through the promise.